```
panic: strings: negative Repeat count

goroutine 1 [running]:
strings.Repeat({0x10446b9f4?, 0x10446dd39?}, 0x60?)
        /Users/nderjung/go/src/strings/strings.go:549 +0x2e0
kraftkit.sh/internal/cli/kraft/cloud/utils.printBar(0x14000468ec7, 0x14000736f30?, 0x1400063b700?)
        /Users/nderjung/Workspace/unikraft/kraftkit-pr-1620/internal/cli/kraft/cloud/utils/print.go:629 +0xb8
kraftkit.sh/internal/cli/kraft/cloud/utils.PrintQuotas({0x104ead358, 0x14000736f30}, {{0x14000320cc0, 0x1b}, {0x14000320cdc, 0x20}, {0x1044884da, 0x11}, 0x1}, {0x104b2fc93, ...}, ...)
        /Users/nderjung/Workspace/unikraft/kraftkit-pr-1620/internal/cli/kraft/cloud/utils/print.go:735 +0x2d84
kraftkit.sh/internal/cli/kraft/cloud/quotas.(*QuotasOptions).Run(0x140003b40c0, {0x104ead358, 0x14000736f30}, {0x0?, 0x0?, 0x0?})
        /Users/nderjung/Workspace/unikraft/kraftkit-pr-1620/internal/cli/kraft/cloud/quotas/quotas.go:97 +0x3fc
kraftkit.sh/cmdfactory.New.func1(0x140002dd808?, {0x10628ee28?, 0x0?, 0x0?})
        /Users/nderjung/Workspace/unikraft/kraftkit-pr-1620/cmdfactory/builder.go:371 +0x4c
kraftkit.sh/cmdfactory.AttributeFlags.bind.func3(0x140002dd808, {0x10628ee28, 0x0, 0x0})
        /Users/nderjung/Workspace/unikraft/kraftkit-pr-1620/cmdfactory/builder.go:553 +0x14c
github.com/spf13/cobra.(*Command).execute(0x140002dd808, {0x10628ee28, 0x0, 0x0})
        /Users/nderjung/go/pkg/mod/github.com/spf13/cobra@v1.8.0/command.go:983 +0x840
github.com/spf13/cobra.(*Command).ExecuteC(0x1400002e908)
        /Users/nderjung/go/pkg/mod/github.com/spf13/cobra@v1.8.0/command.go:1115 +0x344
github.com/spf13/cobra.(*Command).Execute(...)
        /Users/nderjung/go/pkg/mod/github.com/spf13/cobra@v1.8.0/command.go:1039
github.com/spf13/cobra.(*Command).ExecuteContext(...)
        /Users/nderjung/go/pkg/mod/github.com/spf13/cobra@v1.8.0/command.go:1032
kraftkit.sh/cmdfactory.Main({0x104ead358, 0x14000736f30}, 0x1400002e908)
        /Users/nderjung/Workspace/unikraft/kraftkit-pr-1620/cmdfactory/builder.go:183 +0x6c
kraftkit.sh/internal/cli/kraft.Main({0x16d043030?, 0x102dffe40?, 0x10613abe8?})
        /Users/nderjung/Workspace/unikraft/kraftkit-pr-1620/internal/cli/kraft/kraft.go:204 +0x884
main.main()
        /Users/nderjung/Workspace/unikraft/kraftkit-pr-1620/cmd/kraft/main.go:19 +0x94
```

--- a/go.mod
+++ b/go.mod
@@ -79,7 +79,7 @@ require (
 	k8s.io/apimachinery v0.29.3
 	k8s.io/apiserver v0.29.3
 	oras.land/oras-go/v2 v2.5.0
-	sdk.kraft.cloud v0.5.9-0.20240417173611-0430af5a2dac
+	sdk.kraft.cloud v0.5.9-0.20240430124855-83577e6062d6
 	sigs.k8s.io/kustomize/kyaml v0.17.0
 )
 

--- a/internal/cli/kraft/cloud/quotas/quotas.go
+++ b/internal/cli/kraft/cloud/quotas/quotas.go
@@ -74,13 +74,18 @@ func (opts *QuotasOptions) Run(ctx context.Context, _ []string) error {
 		return fmt.Errorf("could not retrieve credentials: %w", err)
 	}
 
-	client := kraftcloud.NewUsersClient(
+	client := kraftcloud.NewClient(
 		kraftcloud.WithToken(config.GetKraftCloudTokenAuthConfig(*auth)),
 	)
 
-	resp, err := client.WithMetro(opts.metro).Quotas(ctx)
+	resp, err := client.Users().WithMetro(opts.metro).Quotas(ctx)
 	if err != nil {
 		return fmt.Errorf("could not get quotas: %w", err)
+	}
+
+	imageResp, err := client.Images().WithMetro(opts.metro).Quotas(ctx)
+	if err != nil {
+		return fmt.Errorf("could not get image quotas: %w", err)
 	}
 
 	if err = iostreams.G(ctx).StartPager(); err != nil {
@@ -89,5 +94,5 @@ func (opts *QuotasOptions) Run(ctx context.Context, _ []string) error {
 
 	defer iostreams.G(ctx).StopPager()
 
-	return utils.PrintQuotas(ctx, *auth, opts.Output, *resp)
+	return utils.PrintQuotas(ctx, *auth, opts.Output, *resp, imageResp)
 }

--- a/internal/cli/kraft/cloud/utils/print.go
+++ b/internal/cli/kraft/cloud/utils/print.go
@@ -24,6 +24,7 @@ import (
 
 	kccerts "sdk.kraft.cloud/certificates"
 	kcclient "sdk.kraft.cloud/client"
+	kcimages "sdk.kraft.cloud/images"
 	kcinstances "sdk.kraft.cloud/instances"
 	kcservices "sdk.kraft.cloud/services"
 	kcautoscale "sdk.kraft.cloud/services/autoscale"
@@ -633,7 +634,7 @@ func printBar(cs *iostreams.ColorScheme, progress, max int) string {
 
 // PrintQuotas pretty-prints the provided set of user quotas or returns
 // an error if unable to send to stdout via the provided context.
-func PrintQuotas(ctx context.Context, auth config.AuthConfig, format string, resp kcclient.ServiceResponse[kcusers.QuotasResponseItem]) error {
+func PrintQuotas(ctx context.Context, auth config.AuthConfig, format string, resp kcclient.ServiceResponse[kcusers.QuotasResponseItem], imageResp *kcimages.QuotasResponseItem) error {
 	if format == "raw" {
 		printRaw(ctx, resp)
 		return nil
@@ -662,6 +663,15 @@ func PrintQuotas(ctx context.Context, auth config.AuthConfig, format string, res
 	// Blank line on list view
 	if format == "list" {
 		table.AddField("", nil)
+	}
+
+	if imageResp != nil && imageResp.Hard > 0 {
+		table.AddField("IMAGE STORAGE", cs.Bold)
+
+		// Blank line on list view
+		if format == "list" {
+			table.AddField("", nil)
+		}
 	}
 
 	table.AddField("ACTIVE INSTANCES", cs.Bold)
@@ -719,6 +729,24 @@ func PrintQuotas(ctx context.Context, auth config.AuthConfig, format string, res
 	// Blank line on list view
 	if format == "list" {
 		table.AddField("", nil)
+	}
+
+	// IMAGE STORAGE
+	if imageResp != nil && imageResp.Hard > 0 {
+		var storedImages string
+		if format == "list" {
+			storedImages = printBar(cs, int(imageResp.Used), int(imageResp.Hard)) + " "
+		}
+		storedImages += fmt.Sprintf("%s/%s",
+			humanize.IBytes(uint64(imageResp.Used)*humanize.Byte),
+			humanize.IBytes(uint64(imageResp.Hard)*humanize.Byte),
+		)
+		table.AddField(storedImages, nil)
+
+		// Blank line on list view
+		if format == "list" {
+			table.AddField("", nil)
+		}
 	}
 
 	// ACTIVE INSTANCES

